code/game/objects/items/explosives/grenades/marines.dm

```diff
@@ -394,25 +394,25 @@
 
 	var/obj/item/card/id/id = user.get_idcard()
 	var/list/turf_list = list()
+	var/cnt = 1
 
-	for(var/turf/T in view(range, loc))
+	for(var/turf/T AS in view(range, loc))
 		turf_list += T
 
-	for(var/i = 1 to turf_list.len)
-		var/turf/U = pick(turf_list)
-		turf_list.Remove(U)
-		if(!U.density && !turf_block_check(src, U) && !(locate(/obj/item/explosive/mine) in range(1, U)))
-			var/obj/item/explosive/mine/placed_mine = new /obj/item/explosive/mine(U)
+	while(placed <= amount && cnt <= turf_list.len)
+		var/turf/target_turf = pick(turf_list)
```

Comment:
no, use pick_n_take and you don't need cnt

Reply:
If i remove cnt it can loop for way longer than it should

Reply:
i said to use pick_n_take :)